torchtext/experimental/transforms.py

```diff
@@ -0,0 +1,154 @@
+import torch
+from torchtext.data.utils import get_tokenizer
+from torchtext.experimental import functional as F
+
+
+class TokenizerTransform(torch.nn.Module):
+    def __init__(self, tokenizer=get_tokenizer('basic_english')):
+        """Initiate Tokenizer transform.
+
+        Arguments:
+            tokenizer: a callable object to convert a text string
+                to a list of token. Default: 'basic_english' tokenizer
+        """
+
+        super(TokenizerTransform, self).__init__()
+        self.tokenizer = tokenizer
+
+    def forward(self, str_input):
+        """
+        Inputs:
+            str_input: a text string
+
+        Outputs:
+            A list of tokens
+
+        Examples:
+            >>> tok_transform = torchtext.experimental.transforms.TokenizerTransform()
+            >>> tok_transform('here we are')
+            >>> ['here', 'we', 'are']
+        """
+        # type: (str) -> List[str]
+        return self.tokenizer(str_input)
+
+
+class VocabTransform(torch.nn.Module):
```

Comment:
It seems like the point here is to map the vocab across a given list of input tokens and then return the list. A user could technically do that herself pretty easily (as you can see by the amount of code). Maybe we can add more value to this by also supporting nested streams of input tokens?